app/src/test/kotlin/compiler/e2e/CorrectProgramsE2eTest.kt

```diff
@@ -215,4 +215,37 @@ class CorrectProgramsE2eTest {
             """
         )
     }
+
+    @Test
+    fun `test generators`() {
+        // TODO: exchange to "assertProgramCorrect" after the next phases are implemented
+        E2eTestUtils.assertNoDiagnosticsOfType(
+            """
+            czynność główna() {
+                przekaźnik mój_generator(a: Liczba = 17) -> Liczba {
+                    zm x: Liczba = 1
+                    dopóki (x <= a) {
+                        przekaż x
+                
+                        x = x + 1
+                
+                        if (x == 42)
+                            zakończ // możliwa tylko forma bez wartości
```

Comment:
Since it's e2e test, how about adding another `przekaż` to that function, for example there? It would be useful later for testing if the function returns control back to the correct yield statement (out of 2)?